src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,69 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a deadline task that has a description and a due date/time.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
+
+    /**
+     * Constructor with description and due date/time.
+     *
+     * @param description The description of the deadline.
+     * @param by The due date/time of the deadline.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = convertDateTime(by);
+    }
+
+    /**
+     * Converts the string by to type LocalDateTime.
+     *
+     * @param dateTimeString String by to be converted.
+     * @return LocalDateTime by.
+     */
+    public LocalDateTime convertDateTime(String dateTimeString) {
```

Comment:
You may consider putting the parsing logic to the parser. I can see your point too.